test/integration/gardenlet/bastion/bastion_test.go

```diff
@@ -0,0 +1,254 @@
+			g.Expect(extBastion.Spec.UserData).To(Equal(createUserData(bastion)))
+		}).Should(Succeed())
+
+		By("Patch the extension Bastion to satisfy the condition for readyness as there is no extension controller running in test")
+		patch := client.MergeFrom(extBastion.DeepCopy())
+		annotations := extBastion.GetAnnotations()
+		delete(annotations, v1beta1constants.GardenerOperation)
+		extBastion.SetAnnotations(annotations)
+		Expect(testClient.Patch(ctx, extBastion, patch)).To(Succeed())
+		extBastion.Status.ObservedGeneration = extBastion.GetGeneration()
+		extBastion.Status.LastOperation = &gardencorev1beta1.LastOperation{
+			LastUpdateTime: metav1.Now(),
+			State:          gardencorev1beta1.LastOperationStateSucceeded,
+		}
+		Expect(testClient.Status().Patch(ctx, extBastion, patch)).To(Succeed())
+
+		Eventually(func(g Gomega) {
+			g.Expect(testClient.Get(ctx, client.ObjectKey{Namespace: shootNamespace.Name, Name: resourceName}, &extensionsv1alpha1.Bastion{})).To(Succeed())
+			g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(bastion), bastion)).To(Succeed())
+			g.Expect(bastion.Status.Conditions).To(ContainElement(MatchFields(IgnoreExtras, Fields{
+				"Type":    Equal(operationsv1alpha1.BastionReady),
+				"Status":  Equal(gardencorev1alpha1.ConditionTrue),
+				"Reason":  Equal("SuccessfullyReconciled"),
+				"Message": Equal("The bastion has been reconciled successfully."),
+			})))
+			g.Expect(bastion.Status.ObservedGeneration).To(Equal(pointer.Int64(bastion.Generation)))
+		}).Should(Succeed())
+	}
+
+	It("should add the finalizer", func() {
+		Eventually(func(g Gomega) {
+			g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(bastion), bastion)).To(Succeed())
+			g.Expect(bastion.Finalizers).To(ConsistOf("gardener"))
+		}).Should(Succeed())
+	})
+
+	Context("#Reconcile", func() {
+		It("should create or patch the bastion in the seed cluster", func() {
+			Eventually(func(g Gomega) {
+				extBastion := &extensionsv1alpha1.Bastion{}
+				g.Expect(testClient.Get(ctx, client.ObjectKey{Namespace: shootNamespace.Name, Name: resourceName}, extBastion)).To(Succeed())
+				annotations := extBastion.GetAnnotations()
+				g.Expect(annotations).To(HaveKeyWithValue(v1beta1constants.GardenerOperation, v1beta1constants.GardenerOperationReconcile))
+				g.Expect(annotations).To(HaveKey(v1beta1constants.GardenerTimestamp))
+				g.Expect(extBastion.Spec.Type).To(Equal(*bastion.Spec.ProviderType))
+				g.Expect(extBastion.Spec.UserData).To(Equal(createUserData(bastion)))
+			}).Should(Succeed())
+		})
+
+		It("should set BastionReady to True once extension Bastion is ready", func() {
+			bastionReady()
+		})
+	})
+
+	Context("#Delete", func() {
+		JustBeforeEach(func() {
+			bastionReady()
+
+			By("Add finalizer to Bastion")
+			patch := client.MergeFrom(bastion.DeepCopy())
+			Expect(controllerutil.AddFinalizer(bastion, testID)).To(BeTrue())
+			Expect(testClient.Patch(ctx, bastion, patch)).To(Succeed())
```

Comment:
See above, this would no longer be needed

Reply:
This finalizer is added to prolong the deletion of bastion object.